### PR TITLE
Changes to allow custom options for filesystems

### PR DESCRIPTION
**Pull request by drewmoseley**
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

I added options to allow for the customization of mount options and mkfs option.  These are required to properly support read-only systems with mender-binary-delta updates.

### CODE REVIEW

File: configs/mender_convert_config [hunk 1]
```diff
@@ -32,9 +32,21 @@ MENDER_ARTIFACT_COMPRESSION="gzip"
...
 MENDER_ENABLE_SYSTEMD=y
 
+
+# Custom mkfs options for creating the rootfs partition
+MENDER_ROOT_PART_MKFS_OPTS=""
+
+# Set the fstab options for mounting the root partition
+MENDER_ROOT_PART_FSTAB_OPTS="defaults"
+# Set to 0 to disable fsck; needed for read-only rootfs support
+MENDER_ROOT_PART_FS_PASSNO="1"
```
Comment (mirzak) [inline configs/mender_convert_config:42]: This can be one configuration option, e.g:

```
MENDER_ROOT_PART_FSTAB_OPTS = "defaults 1 1"
```

**Review by mirzak — CHANGES_REQUESTED** [review]

File: modules/disk.sh [hunk 2]
```diff
@@ -114,21 +114,27 @@ disk_create_file_system_from_folder() {
...
+  esac
+
   case "$4" in
     "ext4")
       MKFS_EXT4="/usr/bin/mkfs.ext4"
       if [ ! -f ${MKFS_EXT4} ]; then
         MKFS_EXT4="/sbin/mkfs.ext4"
       fi
-      run_and_log_cmd "${MKFS_EXT4} -q -F ${2}"
+      run_and_log_cmd "${MKFS_EXT4} -q -F ${2} -i 4096 ${EXTRA_OPTS}"
```
Comment (kacf) [inline modules/disk.sh:129]: The `-i 4096` was snuck into the same commit. Why? If it's intentional I think this should be in a separate commit.

File: configs/mender_convert_config (hunk 1 above)
Comment (kacf) [reply configs/mender_convert_config:42]: Yes, but in Yocto the same variable is used only for the first part, so it's probably better to keep them split for consistency. The dump and pass number aren't configurable in Yocto ATM.

**Review by kacf — CHANGES_REQUESTED** [review]

File: modules/disk.sh (hunk 2 above)
Comment (drewmoseley) [reply modules/disk.sh]: Rebase failure.  It's fixed now

**Review by kacf — APPROVED** [review]
Thanks, looks good!

@mirzak?
